DS-5240 As a SM I want it to be clear to which Revisions user has given Consent

OH17L opened the pull request:
You can see HTT [here](https://github.com/goalgorilla/open_social/pull/877).

bramtenhove committed:
DS-5240 by bramtenhove: Add Views handlers for User Consent entity

ohorbatiuk committed:
Merge branch 'feature/DS-5351-gdpr-improves' of github.com:goalgorilla/gdpr_consent into feature/DS-5240-consent-overview

ohorbatiuk committed:
DS-5240 by chmez: Prepare page with last consent of users.

ohorbatiuk committed:
DS-5240 by chmez:

1. Show data policy revision created date instead of ID and show user consent state label instead of ID on the user consents overview page.

2. Delete user consents table on the data policy revision overview page.

ohorbatiuk committed:
DS-5240 by chmez: Change line separators in the Views view configuration file.

ohorbatiuk committed:
DS-5240 by chmez: Add a message to the empty table.

bramtenhove committed:
DS-5240 by bramtenhove: Textual change

bramtenhove commented:
@OH17L, I found two functional issues.

**1. Last action of user is shown**
The last action of the user is shown, but this is _not correct_! It should be last action of the user **per revision**.

Here I'm Paul and I agreed to the revision of 17:13.
![screen shot 2018-05-14 at 17 25 03](https://user-images.githubusercontent.com/7124754/40007206-6f22bcd8-579c-11e8-9790-f25f8fe7f261.png)

Here I'm Paul and I'm undecided for the revision of 17:21, but I can no longer filter on the previous revision and my action there is not shown.
![screen shot 2018-05-14 at 17 26 28](https://user-images.githubusercontent.com/7124754/40007270-9b0a7d40-579c-11e8-89c8-617cea17e1ad.png)

**2. Filter is incomplete**
The filter with revisions shows only the revision that has user consents. Rather this should show all the revisions who are _active_ or have been _active_.

Next to that we should not have _Any_ as default option the filter, but rather the current active revision.

bramtenhove reviewed (changes requested):
See comment at https://github.com/goalgorilla/gdpr_consent/pull/12#issuecomment-388860790

ohorbatiuk committed:
Merge branch 'develop' of https://github.com/goalgorilla/gdpr_consent into feature/DS-5240-consent-overview

ohorbatiuk committed:
DS-5240 by chmez:

1. Show last user consent per revision.

2. Allow filtering user consents by revisions which are active or was active.
